src/app/state/rest/useApiRequest.ts

```diff
@@ -102,17 +70,14 @@ const useApiRequest = <SCHEMA>({ url, group }: Config) => {
   const execPatch = useCallback((payload: {}) => exec("patch", payload), [ exec ])
   const execDelete = useCallback((payload: {}) => exec("delete", payload), [ exec ])
 
-  // onMount, execute get:
-  useEffect(() => { execGet() }, [ execGet ])
-
-  // onUnmount, cancel tokens:
-  useEffect(() => () => { cancelTokens.current.forEach(_ => _.cancel()) }, [ cancelTokens ])
+  // reFetch whenever our cache is invalidated
+  const data = cache[url]
+  useEffect(() => { if (!data) { execGet() } }, [ execGet, data ])
```

Comment:
I think this is a really cool feature:
If the cache is invalidated while the component using this API-data is still mounted, it will refetch it automatically.

Reply:
If the hook gets called again before the response returns it will be fetched more than once. Cause the cache will only be filled after a successful response.

Reply:
Then it would take the pending request instead of creating a new one.

Reply:
Ah, yes. Then `setItem` will be called more than once? (with the same result)

Reply:
Yes